### ISSUE TITLE
0.12 support

### ISSUE THREAD

**Issue by CodingMadeEasy** [terraform-providers/terraform-provider-nomad#60]
Hi, 

Can you please make a version with 0.12 support? I recently upgraded to use some of the new features, but I sadly am not able to work because there's no 0.12 compatible nomad provider version.

Thanks in advance

**Comment by peimanja** [comment]
Is there any timeline on this? 🥇

**Comment by rmlsun** [comment]
we're blocked by this too

**Comment by ketzacoatl** [comment]
@mitchellh can you help us? We are part of the community and your customer base that wants to use two of your products together (at least!) - We do not see the nomad provider getting the level of attention that it needs. This provider lacks quality and lags behind in support for features coming out of Terraform and Nomad. We are being held back when using these products with our clients, and the situation makes it more difficult to sell Hashicorp products, and we would much rather be successful in selling your products, please help!!

Specifically, the nomad provider:
* does not work with Terraform 0.12.x
* latest release is buggy and crashing
* does not fully support the 0.9.x nomad job spec
* does not actually talk to nomad to find out about current resources and their state, and doesn't update nomad to match what is defined in the Terraform

ATM, we are better off using the `local_exec` module with `nomad plan` and `nomad run` than this provider (though that's painful for plenty of other reasons).

As a community, we've given the dev team patience, support, and understanding over the last 2 years, but how can we continue to make this work?

For community plugins, I would set the bar lower, but this is Nomad and Terraform we're talking about! Why does Hashicorp not ensure this provider shines like gold to exemplify your products together?!

Thanks for listening, please help :)

**Comment by cgbaker** [comment]
@ketzacoatl et al.: I apologize for not addressing this earlier, this GH issue doing through the cracks.

The 1.4.0 Nomad provider released in May supports Terraform 0.12:
https://github.com/terraform-providers/terraform-provider-nomad/blob/master/CHANGELOG.md#140-may-20-2019

Thank you for bringing this up.

**Comment by CodingMadeEasy** [comment]
Thanks for the update. Sadly I already ported over to Pulumi due to the lack of response. I hope to be back on Terraform sometime in the future

**Comment by ketzacoatl** [comment]
I'm mostly trippin on #1, but I'm not quiet understanding @cgbaker, are you saying the provider currently supports TF 0.12.x, but not all of the spec? Or should this ticket be closed?

**Comment by appilon** [comment]
Hello, you should be able to use the provider with version 1.4 of the provider.
```
~/terraform/nomad-test
❯ terraform -v
Terraform v0.12.3-dev

Your version of Terraform is out of date! The latest version
is 0.12.3. You can update by downloading from www.terraform.io/downloads.html

~/terraform/nomad-test
❯ terraform init

Initializing the backend...

Initializing provider plugins...
- Checking for available provider plugins...
- Downloading plugin for provider "nomad" (terraform-providers/nomad) 1.4.0...

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
```

**Comment by shantanugadgil** [comment]
@ketzacoatl the idea about `local-exec` is pretty sweet. Brilliant, even! :grinning: 

I say that because it creates a sort of nice separation between Terraform and the Nomad job; it allows one to do things using Nomad CLI if needed for debugging, etc purposes! :+1:

**Comment by ketzacoatl** [comment]
Glad that's helpful @shantanugadgil!

@cgbaker are we able to close this b/c the provider already supports the 0.12.x attributes, or are there some attributes that are known "need to be added"?

**Comment by cgbaker** [comment]
I don't think there's any other work necessary to support 0.12. if that
turns out not to be the case, we can reopen or file a new issue.
